Source/JavaScriptCore/bytecode/StructureStubInfo.h

```diff
@@ -431,7 +462,7 @@ class StructureStubInfo {
     // Note that it's always safe to clear this. If we clear it prematurely, then if we see the same
     // structure again during this buffering countdown, we will create an AccessCase object for it.
     // That's not so bad - we'll get rid of the redundant ones once we regenerate.
-    HashSet<BufferedStructure, BufferedStructure::Hash, BufferedStructure::KeyTraits> m_bufferedStructures WTF_GUARDED_BY_LOCK(m_bufferedStructuresLock);
+    std::variant<std::monostate, Vector<StructureID>, Vector<std::tuple<StructureID, CacheableIdentifier>>> m_bufferedStructures WTF_GUARDED_BY_LOCK(m_bufferedStructuresLock);
```

Comment:
Why is the mono state case necessary? At first glance it feels like a size-0 vector would serve the same purpose & be just as default-constructible

Reply:
`StructureStubInfo` is initialized before it gets information about AccessType, so at first, it does not know which one is required.